bucket.go

```diff
@@ -95,65 +122,120 @@ func (c *Client) UpdateBucket(id string, options BucketOptions) (MessageResponse
+	}
 	request, err := http.NewRequest(http.MethodPut, c.clientTransport.baseUrl.String()+"/bucket/"+id, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	res, err := c.session.Do(request)
 	if err != nil {
-		panic(err)
+		return MessageResponse{}, err
 	}
 
 	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	var data MessageResponse
-	var error_ BucketResponseError
+	var errResp BucketResponseError
 	err = json.Unmarshal(body, &data)
-	err = json.Unmarshal(body, &error_)
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
+	_ = json.Unmarshal(body, &errResp)
+	if errResp.Errors != "" {
+		return MessageResponse{}, err
+	}
 
-	return data, error_
+	return data, nil
 }
 
-func (c *Client) EmptyBucket(id string) (MessageResponse, BucketResponseError) {
-	jsonBody, _ := json.Marshal(map[string]interface{}{})
+func (c *Client) EmptyBucket(id string) (MessageResponse, error) {
+	jsonBody, err := json.Marshal(map[string]interface{}{})
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	res, err := c.session.Post(c.clientTransport.baseUrl.String()+"/bucket/"+id+"/empty", "application/json", bytes.NewBuffer(jsonBody))
 	if err != nil {
-		panic(err)
+		return MessageResponse{}, err
 	}
 
 	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	var data MessageResponse
-	var error_ BucketResponseError
+	var errResp BucketResponseError
 	err = json.Unmarshal(body, &data)
-	err = json.Unmarshal(body, &error_)
+	if err != nil {
+		return MessageResponse{}, err
+	}
+	_ = json.Unmarshal(body, &errResp)
+	if errResp.Errors != "" {
+		return data, errResp
+	}
 
-	return data, error_
+	return data, nil
 }
 
-func (c *Client) DeleteBucket(id string) (MessageResponse, BucketResponseError) {
-	jsonBody, _ := json.Marshal(map[string]interface{}{})
+func (c *Client) DeleteBucket(id string) (MessageResponse, error) {
+	jsonBody, err := json.Marshal(map[string]interface{}{})
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	request, err := http.NewRequest(http.MethodDelete, c.clientTransport.baseUrl.String()+"/bucket/"+id, bytes.NewBuffer(jsonBody))
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	res, err := c.session.Do(request)
 	if err != nil {
-		panic(err)
+		return MessageResponse{}, err
 	}
 
 	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return MessageResponse{}, err
+	}
+
 	var data MessageResponse
-	var error_ BucketResponseError
+	var errResp BucketResponseError
 	err = json.Unmarshal(body, &data)
-	err = json.Unmarshal(body, &error_)
+	if err != nil {
+		return MessageResponse{}, err
+	}
 
-	return data, error_
+	_ = json.Unmarshal(body, &errResp)
+	if errResp.Errors != "" {
+		return data, errResp
+	}
+
+	return data, nil
 }
 
 type MessageResponse struct {
 	Message string `json:"message"`
 }
 
 type BucketResponseError struct {
-	Error      string `json:"error"`
+	Errors     string `json:"error"`
```

Comment:
I believe it is better to retain it as error because, given the multitude of errors, it should be an array of strings or structs.